src/common/applyLanguage.ts

```diff
@@ -4,13 +4,39 @@ import { Language, defaultDetokenizer, Plugin } from "./Language";
 import { programToSpine } from "./Spine";
 import polygolfLanguage from "../languages/polygolf";
 
+export type OptimisationLevel = "none" | "heuristic" | "full";
```

Comment:
"heuristic" doesn't seem to be handled. Put in TODO comment, or maybe this is more appropriate as a GH issue.